cdap-app-fabric/src/main/java/io/cdap/cdap/internal/namespace/credential/handler/GcpWorkloadIdentityHttpHandler.java

```diff
@@ -0,0 +1,267 @@
+   * Validates a credential identity.
+   *
+   * @param request   The HTTP request.
+   * @param responder The HTTP responder.
+   * @throws BadRequestException If identity validation fails.
+   * @throws NotFoundException   If the associated profile is not found.
+   * @throws IOException         If transport errors occur.
+   */
+  @POST
+  @Path("/namespaces/{namespace-id}/credentials/workloadIdentity/validate")
+  public void validateIdentity(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace) throws Exception {
+    accessEnforcer.enforce(new NamespaceId(namespace), NamespacePermission.PROVISION_CREDENTIAL);
+    NamespaceWorkloadIdentity namespaceWorkloadIdentity =
+        deserializeRequestContent(request, NamespaceWorkloadIdentity.class);
+    if (Strings.isNullOrEmpty(namespaceWorkloadIdentity.getIdentity())) {
+      throw new BadRequestException("Identity cannot be null or empty.");
+    }
+    NamespaceMeta namespaceMeta = getNamespaceMeta(namespace);
+    validateNamespaceIdentity(namespaceMeta, namespaceWorkloadIdentity);
+    CredentialIdentity credentialIdentity = new CredentialIdentity(
+        NamespaceId.SYSTEM.getNamespace(), GcpWorkloadIdentityUtil.SYSTEM_PROFILE_NAME,
+        namespaceWorkloadIdentity.getIdentity(),
+        namespaceWorkloadIdentity.getServiceAccount());
+    switchToInternalUser();
+    try {
+      credentialProvider.validateIdentity(namespaceMeta, credentialIdentity);
+    } catch (IdentityValidationException e) {
+      throw new BadRequestException(String.format("Identity validation failed with error: %s",
+          e.getMessage()), e);
+    } catch (io.cdap.cdap.proto.credential.NotFoundException e) {
+      throw new NotFoundException(e.getMessage());
+    }
+    responder.sendJson(HttpResponseStatus.OK, "Namespace identity validated successfully");
+  }
+
+  /**
+   * Fetches a credential identity.
+   *
+   * @param request      The HTTP request.
+   * @param responder    The HTTP responder.
+   * @param namespace    The identity namespace.
+   * @throws BadRequestException If the identity name is invalid.
+   * @throws IOException         If transport errors occur.
+   * @throws NotFoundException   If the namespace or identity are not found.
+   */
+  @GET
+  @Path("/namespaces/{namespace-id}/credentials/workloadIdentity")
+  public void getIdentity(HttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace) throws Exception {
+    NamespaceMeta namespaceMeta = getNamespaceMeta(namespace);
+    CredentialIdentityId credentialIdentityId = createIdentityIdOrPropagate(namespace,
+        GcpWorkloadIdentityUtil.getWorkloadIdentityName(namespaceMeta.getIdentity()));
+    switchToInternalUser();
+    Optional<CredentialIdentity> identity = credentialIdentityManager.get(credentialIdentityId);
+    if (!identity.isPresent()) {
+      throw new NotFoundException("Namespace identity not found.");
+    } else {
```

Comment:
nit: We can omit the else for readability.

Reply:
done